action.yml

```diff
@@ -0,0 +1,85 @@
+name: 'silverbackltd-build-action'
+author: 'Silverback Ltd.'
+description: 'A GitHub Action to build docker images for a silverback repository and push to a container registry'
+branding:
+  icon: 'check-circle'
+  color: 'green'
+
+outpus:
+  built_images:
+    description: 'List of Docker images that were built and pushed.'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set Environment Variables
+      shell: bash
+      run: |
+        echo "REGISTRY=ghcr.io" >> $GITHUB_ENV
+        echo "IMAGE_NAME=${{ github.repository }}"
+        echo "TARGET_DIR=$GITHUB_WORKSPACE/.silverback-images"
+        echo "DOCKERFILE_PATTERN=Dockerfile.*"
+        echo "GITHUB_EVENT_NAME=${{ github.event_name }}"
+        echo "GITHUB_REF=${{ github.ref }}"
+
+    - name: Check if Directory Exists
+      id: check_dir
+      run: |
+        if [ -d "${{ env.TARGET_DIR }}" ]; then
+          echo "Directory exists."
+          echo "directory_exists=true" >> $GITHUB_OUTPUT
+        else
+          echo "Directory does not exist. Creating..."
+          mkdir -p "${{ env.TARGET_DIR }}"
+          echo "directory_exists=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Set Lowercase Repository Owner
+      id: lowercase_owner
+      run: |
+        LOWERCASE_OWNER=$(echo "${GITHUB_REPOSITORY_OWNER}" | tr '[:upper:]' '[:lower:]')
+        echo "LOWERCASE_OWNER=$LOWERCASE_OWNER" >> $GITHUB_ENV
+
+    - name: Generate Files
+      id: gen_files
+      if: steps.check_dir.outputs.directory_exists == 'false'
+      run: |
+        pip install silverback
+        silverback build --generate
+
+    - name: Verify Dockerfiles Exist
+      run: |
+        if [ ! -d "${{ env.TARGET_DIR }}" ]; then
+          echo "Directory '${{ env.TARGET_DIR }}' does not exist. Exiting."
+          exit 1
+        fi
+
+    - name: Log into GitHub Container Registry
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I was thinking that the login and publish part could be done separately. It should at the very least be user configurable when to publish.

Reply:
Agreed, and then I can allow for configure on push. I've been working through as many ideas I can think of. i think I have a good option here. Should be out by eod